opennms-model/src/main/java/org/opennms/netmgt/model/LldpLinkTopologyEntity.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.model;
+
+import java.util.Objects;
+
+import org.opennms.core.utils.LldpUtils;
+
+/**
+ * This is NOT a Hibernate/JPA entity but rather a lightweight model without less attributes than LldpLink and no lazy
```

Comment:
I am not sure I understand what problem we want to address. we can add an annotation. But what for? Be a marker annotation? The class is not marked with the Hibernate / JPA annotation Entity and thus can't be saved with Hibernate / JPA.

Reply:
All entities now get the same copied comment. So I thought it might be a good idea to just create an annotation, move the docuemtnation there and just annotate the class instead.

Reply:
agree, I don't like copying the documentation either. Interface or annotation?

Reply:
I'd like an annotation, e.g. `@ReadOnlyEntity`. I don't know a good name out of the air.

Reply:
hm I am wondering if an interface wouldn't be more versatile since the TopologyEntities have things in common and we might later access these via the interface in a generic context such as getId() but right now we don't use that.

Reply:
as discussed offline: I will add an annotation that marks a class as a read only view on the database

Reply:
@mvrueden annotation is added